pkgs/by-name/ko/komac/package.nix

```diff
@@ -29,34 +32,47 @@ rustPlatform.buildRustPackage {
 
   nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [
-    dbus
-    openssl
-    zstd
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    darwin.apple_sdk.frameworks.Security
-    darwin.apple_sdk.frameworks.SystemConfiguration
-  ];
+  buildInputs =
+    [
+      dbus
+      openssl
+      zstd
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.SystemConfiguration
+    ];
 
   env = {
     OPENSSL_NO_VENDOR = true;
     YRX_REGENERATE_MODULES_RS = "no";
     ZSTD_SYS_USE_PKG_CONFIG = true;
   };
 
-  passthru.tests.version = testers.testVersion {
-    inherit version;
+  doInstallCheck = true;
```

Comment:
I expected this to be set by default for all packages, even through the documentation only describes `buildPythonPackage` and `buildPythonApplication`.

https://github.com/NixOS/nixpkgs/blob/538f39e5cf36264df0668c96f9067c8ee99db13b/doc/hooks/versionCheckHook.section.md?plain=1#L25

Will avoiding default be better for this value?

https://github.com/NixOS/nixpkgs/blob/538f39e5cf36264df0668c96f9067c8ee99db13b/pkgs/stdenv/generic/make-derivation.nix#L199-L205
https://github.com/NixOS/nixpkgs/blob/538f39e5cf36264df0668c96f9067c8ee99db13b/pkgs/top-level/config.nix#L53-L55
https://github.com/NixOS/nixpkgs/blob/538f39e5cf36264df0668c96f9067c8ee99db13b/pkgs/build-support/rust/build-rust-package/default.nix#L98
https://github.com/NixOS/nixpkgs/blob/538f39e5cf36264df0668c96f9067c8ee99db13b/pkgs/development/compilers/rust/make-rust-platform.nix#L21

Reply:
It is not ran by default when building rust apps, so we need to enable `doInstallCheck` so `versionCheckHook` can be ran.

with `doInstallCheck = true`
```
test types::short_description::__nutype_ShortDescription__::should_have_valid_default_value ... ok
test types::tag::__nutype_Tag__::should_have_consistent_len_char_boundaries ... ok

test result: ok. 370 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

Finished cargoCheckHook
checkPhase completed in 2 minutes 24 seconds
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing cargoInstallHook
Finished cargoInstallHook
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0
shrinking /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0/bin/komac
checking for references to /build/ in /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0...
patching script interpreter paths in /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0
stripping (with command strip and flags -S -p) in  /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0/bin
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Executing versionCheckPhase
Did not find version 2.6.0 in the output of the command /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0/bin/komac --help
A manifest creator for winget-pkgs

Usage: komac <COMMAND>

Commands:
  new            Create a new package from scratch
  update         Add a version to a pre-existing package
  remove         Remove a version from winget-pkgs
  cleanup        Finds branches from the fork of winget-pkgs that have had a merged or closed pull request to microsoft/winget-pkgs from them, prompting for which ones to delete [aliases: >
  token          
  list-versions  Lists all versions for a given package [aliases: list]
  show           Output the manifests for a given package and version
  sync-fork      Merges changes from microsoft/winget-pkgs into the fork repository [aliases: sync, merge-upstream]
  complete       Outputs an autocompletion script for the given shell [aliases: autocomplete]
  analyse        Analyses a file and outputs information about it
  help           Print this message or the help of the given subcommand(s)

Options:
  -v, --version  
  -h, --help     Print help
Successfully managed to find version 2.6.0 in the output of the command /nix/store/n0l5i6qwz624yljv645cb6p081lbg33x-komac-2.6.0/bin/komac --version
komac 2.6.0
Finished versionCheckPhase
no Makefile or custom installCheckPhase, doing nothing
```

without it

```
test types::short_description::__nutype_ShortDescription__::should_have_valid_default_value ... ok
test types::tag::__nutype_Tag__::should_have_consistent_len_char_boundaries ... ok

test result: ok. 370 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

Finished cargoCheckHook
checkPhase completed in 2 minutes 23 seconds
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing cargoInstallHook
Finished cargoInstallHook
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/26p6v8j1cpmd80wvhvivkdfg43naf25v-komac-2.6.0
shrinking /nix/store/26p6v8j1cpmd80wvhvivkdfg43naf25v-komac-2.6.0/bin/komac
checking for references to /build/ in /nix/store/26p6v8j1cpmd80wvhvivkdfg43naf25v-komac-2.6.0...
patching script interpreter paths in /nix/store/26p6v8j1cpmd80wvhvivkdfg43naf25v-komac-2.6.0
stripping (with command strip and flags -S -p) in  /nix/store/26p6v8j1cpmd80wvhvivkdfg43naf25v-komac-2.6.0/bin
```

Reply:
I didn't know that, thank you for adding! 🙏